hayeon/baekjoon/정렬/10814-나이순-정렬.js

```diff
@@ -0,0 +1,39 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  // const test = +input[0];
+  const members = input
+    .slice(1, input.length)
+    .map((member) => ({
+      age: +member.split(" ")[0],
+      name: member.split(" ")[1],
+    }))
+    .sort((a, b) => a.age - b.age)
+    .map((member) => String(member.age) + " " + member.name);
+
+  console.log(members.join("\n"));
```

Comment:
```suggestion
  let result = "";
  
  const members = input
    .slice(1)
    .map((member) => {
      const [age, name] = member.split(" ");
      return { age: +age, name };
    })
    .sort((a, b) => a.age - b.age)
    .forEach(({ age, name }) => {
      result += `${age} ${name}\n`
    });

  console.log(result);
```
- 풀이 잘봤습니다~ 🙇‍♂️
- slice에서는 현재 지점부터 배열 끝까지 짜른다면 2번째 인자의 input.length는 필요없습니다~!!
- 첫 번재 map에서는 매번 split을 2번씩 하는 것보다 1번하고, 이를 구조분해할당으로 활용하는게 어떨까요??
- 2번째 map에서 애초에 result한테 값을 누적시켜주는건 어떨까요? 마지막 콘솔에서 join까지 추가 연산 하실 필요가 없어집니다
- 그리고 굳이 String + " " + mamber.name으로 하신 이유가 있으실까요? 기존 풀이들처럼 `템플릿 리터럴(``)`을 활용하시는게 더 깔끔해보입니다 🙇‍♂️
- 참고로 `+`연산자를 제외한 나머지 사칙연산 연산자들은 문자열로 되있는 숫자("123", "23", "5629" 등등)를 자동으로 숫자로 형변환해줘서 사실 +숫자 처럼 형변환을 안하셔도 결과는 동일합니다 !
- 그리고 이를 종합 + 아래처럼 배열만으로도 풀이가 가능합니다 🙆‍♂️
```js
  let result = "";

  input
    .slice(1)
    .map((member) => member.split(" "))
    .sort((a, b) => a[0] - b[0])
    .forEach(([age, name]) => {
      result += `${age} ${name}\n`;
    });

  console.log(result);
```

Reply:
우왓 훨씬 좋네요!!!!!!  좀 어려워서 구글링하며 풀었는데.. 이 문제는 제방식대로 써보는게 좀 힘들었숩니다.... 이문제 다시 풀어보겠습니다!!!!🙇‍♀️